x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/LogicalPlanBuilder.java

```diff
@@ -105,8 +105,11 @@ public LogicalPlan visitQueryNoWith(QueryNoWithContext ctx) {
         if (limitClause != null) {
             Token limit = limitClause.limit;
             if (limit != null && limitClause.INTEGER_VALUE() != null) {
-                plan = new Limit(source(limitClause), new Literal(source(limitClause),
-                        Integer.parseInt(limit.getText()), DataTypes.INTEGER), plan);
+                if (plan instanceof Limit) {
+                    throw new ParsingException(source(limitClause), "Cannot use both TOP and LIMIT in the same query");
```

Comment:
I would rephrase the error a bit - `TOP and LIMIT are not allowed in the same query - use one or the other`.